src/Dialect/ONNX/DialectBuilder.hpp

```diff
@@ -44,6 +44,8 @@ struct OnnxBuilder : DialectBuilder {
   mlir::Value add(mlir::Value A, mlir::Value B) const;
 
   // ONNXCastOp
+  mlir::Value cast(mlir::Value input, mlir::Value output,
+      mlir::IntegerAttr saturate, mlir::TypeAttr to) const;
```

Comment:
It's weird to pass in an output value to construct a op. `mlir::Value output` should be replaced by `mlir::Type outputType`, and change the order of arguments, say `(mlir::Type outputType, mlir::Value input, mlir::IntegerAttr saturate, mlir::TypeAttr to)`

Reply:
@tungld @imaihal `outputType` is already defined here:

src/Dialect/ONNX/Transforms/Decompose.cpp lines 942 and 943
```
  // The output type will be the same as the target_type or the second input
    Type outputType = target.getType().cast<ShapedType>().getElementType();
```

It seems like we need outputType for src/Dialect/ONNX/DialectBuilder.cpp line 48, if that is the case please add outputType to the construct for cast and remove lines 942 and 943.

Reply:
@hamptonm1 Thanks for the comments. The `outputType` is the type of second input(`target_type`) of CastLike op. It might be better to use `targetType` instead of `outputType`.  We want to use output of CastLikeOp. 
```
Type targetType = target.getType().cast<ShapedType>().getElementType();
Type outputType = output.getType().cast<ShapedType>().getElementType();
```
So I will create this outputType and pass it to builder.

Reply:
Done.

Reply:
@imaihal According to the documentation, `The (first) input tensor will be cast to produce a tensor of the same type as this (second input) tensor.`

So the `targetType` and the `outputType` should be the same type. That is maybe why there is confusion. The tensor types need to be the same for castLike to work correctly.

Reply:
But thanks so much for your update!